src/screens/MyWorkouts/WorkoutSearchBar.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useEffect, useState, useRef } from 'react';
+import { View, Text, TouchableOpacity } from 'react-native';
+import { ScrollView, TextInput } from 'react-native-gesture-handler';
+import { supabase } from '../../utils/supabaseClient';
+
+/**
+ * @param selectedExercise.exercise //value set to exercise pressed
+ * @param selectedExercise.isOk //if true, then value of exercise is valid, otherwise disregard
+ */
+export type SearchBarWorkoutProps = {
+  selectedExercise: { exercise: string | undefined; isOk: boolean | undefined };
+};
+
+const SearchBarWorkout: React.FC<SearchBarWorkoutProps> = (prop: SearchBarWorkoutProps) => {
```

Comment:
Could be nice to de-structure the props in the function definition like `({ onSelectExercise })`